include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -302,6 +302,16 @@ static_assert(__cplusplus >= 201703L, "The Range support requires C++17 as minim
 #    if !defined(_ONEDPL_COMPILE_KERNEL)
 #        define _ONEDPL_COMPILE_KERNEL 1
 #    endif
+
+// Get access to __SYCL_COMPILER_VERSION macro
+#    include <CL/sycl/version.hpp>
+
+// 20201214 value corresponds to oneAPI C++ Compiler Classic 2021.1.2 Patch release
+#    if __SYCL_COMPILER_VERSION <= 20201214
```

Comment:
I suggest using more readable logic: "if we have new compiler we use new API, else - old API".
(in other words, to changes "condition" and "define's order")